### ISSUE TITLE
Reconsider getting rid of layout on ActionGroup

### ISSUE THREAD

**Issue by asangma** [Esri/calcite-design-system#7438]
### Check existing issues

- [X] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates

### Description

By deprecating layout on ActionGroup it removes the ability to use grid layout anywhere but ActionPad.

I suggest either retaining it in ActionGroup or adding `layout=grid` to ActionBar.

### Acceptance Criteria

The ability to do grid layouts without having to use ActionPad.

### Relevant Info

A widget in Maps SDK needs grid layout but not the styles that come with ActionPad.

### Which Component

ActionGroup, ActionBar, ActionPad

### Example Use Case

A widget that needs grid layout but not ActionPad styles.

### Priority impact

p3 - want for upcoming milestone

### Calcite package

- [ ] @esri/calcite-components
- [ ] @esri/calcite-components-react

### Esri team

ArcGIS Maps SDK for JavaScript

**Comment by driskull** [comment]
I think we should decide if `calcite-action-group` should be allowed to be used without a parent component. That will kind of drive whether or not layout is removed or kept.

**Comment by asangma** [comment]
@driskull That's a good point. Also it will help decide whether or not `grid` is available as a layout option on ActionBar.

As of now, in order to make the UI work in the SDK widget, we would be using ActionGroup without an ActionPad or ActionBar parent.

**Comment by SkyeSeitz** [comment]
I lean towards setting grid on the parent to avoid having to use `calcite-action-group` outside of a Pad or Bar.  Makes sense to get rid of `calcite-action-group` and instead add `layout=grid` to ActionBar (would love to see some examples if you got em).

**Comment by ashetland** [comment]
Just to clarify, we're not talking about removing `calcite-action-group`, right? `grid` was a part of `group`, but it was removed in favor of setting the layout at the parent level. Agree with that logic and would definitely like to see some examples of how this played out in Action Bar when a group could be set to a grid layout. A path forward I could see is not adding `grid` to Action Bar (to keep it a single row/column bar), but adding the option to remove the styling from Action Pad.

**Comment by ashetland** [comment]
Scratch the above [comment](https://github.com/Esri/calcite-design-system/issues/7438#issuecomment-1668501489). To clarify agian:

- keep `action-bar-group`
- add `layout=grid` to Action Bar
- layout is set at parent level
- 🚀

**Comment by macandcheese** [comment]
I'd like to bump the priority on this because it prevents many layouts that require this. It's related to #7496 so we can likely consolidate into one.

**Comment by geospatialem** [comment]
> I'd like to bump the priority on this because it prevents many layouts that require this. It's related to #7496 so we can likely consolidate into one.

It looks like this issue is assigned for design work in May. WDYT about using this in favor of #7496 and bumping priority to medium, @macandcheese?

**Comment by macandcheese** [comment]
Sounds good. I think based on above comments design work is done, cc @ashetland

**Comment by ashetland** [comment]
Yep, added a "Design recommendations" section to the top that links to the appropriate comment.

**Comment by josercarcamo** [comment]
@jcfranco @ashetland @driskull @macandcheese I don't believe this issue is ready for development. I've asked a few of the team members for clarification and each has provided a different answer. Could we please create clear requirements for this issue?

**Comment by ashetland** [comment]
Design to do:

- [ ] Flush out specs for `layout="horizontal-fill"` taking into account the collapse button
- [ ] Determine next steps to support `layout="grid" in Action Bar, if needed

**Comment by driskull** [comment]
> I think we could achieve Alan's designs by introducing the "grid" layout to Action Bar (like it exists in Pad) 

I'm not sure grid makes sense in action-bar. A bar by name should only be a single row or column.

> and perhaps adding a "horizontal-stretch" or "horizontal-fill" layout that places actions horizontally, but stretches them to the container width. 

Yeah, I think this makes the most sense to achieve what @asangma is looking for.

> We could potentially add this to Pad as well. However... this does lead to Pad / Bar being basically the same with a visual and perhaps "fill height / width" difference. Maybe an opportunity to combine them in some way to simplify the story...

I'm not sure there's a need to stretch or fill action pad actions as the pad only takes up the space it needs.

**Comment by ashetland** [comment]
Added recommendations from the Calcite team to the issue description.

**Comment by driskull** [comment]
> This will need to force slotted Actions to alignment=“center” and flex to fill

If we modify slotted actions and set them to `alignment="center"` we may be overriding what a user set. I'm not sure if we should be doing that if we want to honor what the user defined for alignment.

The same goes for stretching actions. Should we let the user decide to do that instead of automatically controlling it? If we automatically control it, we need to handle the cases where the action is moved into an action-menu and we may not want it stretched in there.

I think the above need to be considered before proceeding. @asangma @ashetland

**Comment by ashetland** [comment]
The design recommendation would be to center align them, but if we need to rely on docs for that, I think that's okay. We'll default them to center in the ui kit.

**Comment by driskull** [comment]
> The design recommendation would be to center align them, but if we need to rely on docs for that, I think that's okay. We'll default them to center in the ui kit.

I think its probably also a development discussion to figure out if we should be applying the alignment property automatically and whether we should be removing it when actions are slotted into an action-menu automatically. cc @jcfranco

**Comment by jcfranco** [comment]
IIRC, we usually leave public props as-is and rely on doc for recommended configurations. Setting internal props of children through a parent prop is another option, but not applicable here since we're dealing with existing public props.

**Comment by driskull** [comment]
> IIRC, we usually leave public props as-is and rely on doc for recommended configurations. Setting internal props of children through a parent prop is another option, but not applicable here since we're dealing with existing public props.

Agreed.

We may need to add an event on action bar to notify users when an action is moved into an overflow menu so they can remove any props they may need to. Such as `alignment`.

**Comment by driskull** [comment]
Though PR discussion, we've decided to:

- Not make any changes to `action-bar`
- Add a new property `width` to action with values of `"auto" | "full"`
- Users can set `width` on their actions as well as `alignment`.
